play-routes-compiler-cli/CommandLinePlayRoutesCompiler.scala

```diff
@@ -0,0 +1,166 @@
+package rulesplayroutes.routes
+
+import com.google.devtools.build.buildjar.jarhelper.JarCreator
+import higherkindness.rules_scala.common.error.AnnexWorkerError
+import higherkindness.rules_scala.common.worker.WorkerMain
+import higherkindness.rules_scala.common.sandbox.SandboxUtil
+import java.io.{File, PrintStream}
+import java.nio.file.{Files, Path, Paths}
+import play.routes.compiler._
+import play.routes.compiler.RoutesCompiler.RoutesCompilerTask
+import scala.collection.mutable.ListBuffer
+import scala.jdk.CollectionConverters._
+import scala.io.Source
+import scala.util.{Try, Success, Failure}
+import scopt.Read
+import scala.Console._
+import scopt.OptionParser
+
+object CommandLinePlayRoutesCompiler extends WorkerMain[Unit] {
+  implicit val listPathRead: Read[List[Path]] = Read.reads { commaList =>
+    commaList.split(',').view.map(Paths.get(_)).toList
+  }
+
+  case class Config(
+    var sources: List[Path] = List.empty[Path],
+    var outputDirectory: Path = Paths.get("."),
+    var outputSrcJar: Path = Paths.get("."),
+    var additionalImports: List[String] = List.empty[String],
+    var routesGenerator: RoutesGenerator = InjectedRoutesGenerator,
+    var generateReverseRouter: Boolean = false,
+    var namespaceReverserRouter: Boolean = false,
+    var generateForwardsRouter: Boolean = true
+  )
+
+  def parser(workDir: Path) = new OptionParser[Config]("scopt") {
+    head("Command Line Play Routes Compiler", "0.1")
+
+    arg[Path]("<outputDirectory>").required().action { (outputDirectory, config) =>
+      config.outputDirectory = SandboxUtil.getSandboxPath(workDir, outputDirectory)
+      config
+    }.text("directory to output compiled routes to")
+
+    arg[Path]("<outputSrcJar>").required().action { (outputSrcJar, config) =>
+      config.outputSrcJar = SandboxUtil.getSandboxPath(workDir, outputSrcJar)
+      config
+    }.text("file to output srcjar containing compiled routes to")
+
+    arg[List[Path]]("<source1>,<source2>...").unbounded().required().action { (sources, config) =>
+      config.sources = sources.map(SandboxUtil.getSandboxPath(workDir, _))
+      config
+    }.text("routes to compile")
+
+    opt[String]('i', "routesImport").valueName("<import>").unbounded().action { (routesImport, config) =>
+      config.additionalImports = config.additionalImports ++ List(routesImport)
+      config
+    }.text("Imports for the router")
+
+    opt[String]('g', "routesGenerator").valueName("<generator>").maxOccurs(1).action { (routesGeneratorClassString, config) =>
+      config.routesGenerator = {
+        try {
+          val name = if (routesGeneratorClassString.endsWith("$")) {
+            routesGeneratorClassString
+          } else {
+            routesGeneratorClassString + "$"
+          }
+          val clazz = java.lang.Class.forName(name, true, getClass.getClassLoader)
```

Comment:
Changed

I didn't think currentLoader meant `where currentLoader denotes the defining class loader of the current class` because I didn't read the docs closely.

Reply:
Yeah, they can be confusing.